src/Hl7.Fhir.Core/Serialization/SerializationUtil.cs

```diff
@@ -0,0 +1,485 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Xml.Linq;
+using System.Xml.Schema;
+using System.Xml;
+using System;
+using System.Text.RegularExpressions;
+using System.Text;
+using Hl7.Fhir.Utility;
+
+namespace Hl7.Fhir.Serialization
+{
+    public class SerializationUtil
+    {
+        public static bool ProbeIsXml(string data)
+        {
+            Regex xml = new Regex("^<[^>]+>");
+
+            return xml.IsMatch(data.TrimStart());
```

Comment:
can this be written without `TrimStart`? That can allocate a new string for the entire payload, which in Hendrix can get quite large (we accept requests up to 100 MB and regularly see FHIR bundles in the 10s MB range).

Maybe just changing the regex to `^\s*<[^>]+>` would work?

Reply:
OK, done.

(I do hope though that Henderix does not process multi-megabyte FHIR payloads loading them in memory as strings)

Reply:
That is exactly what Hendrix does ... we admittedly could do some things more efficiently. But it works, so 🤷